server/src/tcp/tcp_socket.rs

```diff
@@ -0,0 +1,126 @@
+use tokio::net::TcpSocket;
+
+use crate::configs::tcp::TcpSocketConfig;
+
+pub fn build(config: TcpSocketConfig) -> TcpSocket {
```

Comment:
This method could be called when `enabled = true` - otherwise, we could just use the default values.